plugin/src/main/scala/com/typesafe/genjavadoc/AST.scala

```diff
@@ -97,14 +97,16 @@ trait AST { this: TransformCake ⇒
     if (m.isPublic) "public"
     else if (m.isProtected && !topLevel) "protected"
     else if (m.isPrivate && !topLevel) "private"
-    else "public" // this is the case for “private[xy]” and top level classes
+    else if (m.privateWithin != tpnme.EMPTY) ""
```

Comment:
`privateWithin` may point to *any* enclosing definition (`this`, enclosing class or package, and packages nest). I think you'd need to consider the following mappings:

```
package org.example

class Outer {
  class Inner {
     private[this] val x
     private[Inner] val y
     private [Outer] val z
     private [example] val t
     private [org] val u
  }
}
```

- `private[this]` --> `private`
- `private[Inner]` --> `private`
- `private[Outer]` --> ?? (I guess `private` again)
- `private[org|example]` --> `""`

What do you think?